File: hu.bme.mit.incqueryd.core/hu.bme.mit.incqueryd.engine/src/main/java/hu/bme/mit/incqueryd/engine/rete/messages/ReteCommunicationMessage.java
/*******************************************************************************
 * Copyright (c) 2010-2014, Gabor Szarnyas, Istvan Rath and Daniel Varro
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 * Gabor Szarnyas - initial API and implementation
 *******************************************************************************/
package hu.bme.mit.incqueryd.engine.rete.messages;

import java.io.Serializable;

import scala.collection.immutable.Stack;
import akka.actor.ActorRef;

/**
 * 
 * @author szarnyasg
 * 
 */
public abstract class ReteCommunicationMessage implements Serializable {

	private static final long serialVersionUID = 1L;
	protected Stack<ActorRef> route;

	public ReteCommunicationMessage(final Stack<ActorRef> route) {
		super();
		this.route = route;
	}

	public Stack<ActorRef> getRoute() {
		return route;
	}

	@Override
	public boolean equals(final Object o) {
		if (!(o instanceof ReteCommunicationMessage))
			return false;
		final ReteCommunicationMessage readyMessage = (ReteCommunicationMessage) o;
		
		return route.equals(readyMessage.getRoute());
	}
}

File: hu.bme.mit.incqueryd.client/hu.bme.mit.incqueryd.rete.actors/src/main/scala/hu/bme/mit/incqueryd/engine/rete/messages/ReteCommunicationMessage.scala
package hu.bme.mit.incqueryd.engine.rete.messages

import akka.actor.ActorRef
import hu.bme.mit.incqueryd.engine.rete.dataunits.ChangeSet
import hu.bme.mit.incqueryd.engine.rete.dataunits.ReteNodeSlot

sealed trait ReteCommunicationMessage {
  def route: List[ActorRef]
}

case class TerminationMessage(route: List[ActorRef]) extends ReteCommunicationMessage

case class UpdateMessage(changeSet: ChangeSet, nodeSlot: ReteNodeSlot, route: List[ActorRef]) extends ReteCommunicationMessage